platform/state/platform_state.proto

```diff
@@ -0,0 +1,261 @@
+/**
+ * PlatformState
+ *
+ * Describes the current state of the platform entity.
+ *
+ * The state MUST contain address books, round information,
+ * event hashes, consensus snapshots, and software versions.
+ * The state SHALL accurately represent the latest round's
+ * consensus. <br>
+ * MUST ensure consistency and provide critical data for restarts
+ * and reconnects.
+ *
+ * The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT",
+ * "SHOULD", "SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this
+ * document are to be interpreted as described in [RFC2119](https://www.ietf.org/rfc/rfc2119)
+ * and clarified in [RFC8174](https://www.ietf.org/rfc/rfc8174).
+ */
+syntax = "proto3";
+
+package com.hedera.hapi.platform.state;
+
+/*
+ * Copyright (C) 2024 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+option java_package = "com.hedera.hapi.platform.state.legacy";
+// <<<pbj.java_package = "com.hedera.hapi.platform.state">>> This comment is special code for setting PBJ Compiler java package
+option java_multiple_files = true;
+
+import "basic_types.proto";
+import "timestamp.proto";
+
+/**
+ * PlatformState MUST represent the current consensus state of the platform entity.
+ * The state SHOULD contain the following components:
+ */
+message PlatformState {
+    /**
+     * The address book for this round.
+     *
+     * The address book MUST represent the latest consensus view of node
+     * addresses.
+     */
+    proto.NodeAddressBook address_book = 1;
+
+    /**
+     * The previous address book.
+     *
+     * A temporary workaround until dynamic address books are supported.
+     * If present, the previous address book SHOULD reflect the address
+     * book from the preceding round.
+     */
+    proto.NodeAddressBook previous_address_book = 2;
```

Comment:
It is unfortunate that we could not finish the Dynamic Address Book data structure changes before creating the platform state in protobuf.   We are moving to the ability to support a `queue` of address books, potentially 1 for each round for N number of rounds. 

We are also renaming the platform concept of an address book to `ConsensusRoster` or `Roster`.    And the format for the roster changes drastically after we switch to Dynamic Address Book phase 3.

Reply:
Looks like services will also need to change drastically.

Reply:
Given the fact that I need to have `PlatformState` representation in PBJ, how would you recommend me to proceed?

Reply:
I have an idea, will write up shortly, along with some recommended adjustments to the specification comments.

Reply:
The later address book changes made my idea infeasible.
Most likely we'll just need to deprecate these address book fields when the "roster" upgrade is designed.

Reply:
The good news is that the current plan does not require the address book/roster to be in the platform state at all, and so this is something we can perhaps punt down the road a little while. Current plan will keep all rosters in a map (required for TSS).